app/controllers/candidate_interface/decoupled_references/review_controller.rb

```diff
@@ -11,29 +11,15 @@ def show
       end
 
       def unsubmitted
-        @submit_reference_form = Reference::RefereeSubmitForm.new
+        @submit_reference_form = Reference::SubmitRefereeForm.new
       end
 
       def submit
-        @submit_reference_form = Reference::RefereeSubmitForm.new(submit: submit_param)
+        @submit_reference_form = Reference::SubmitRefereeForm.new(submit: submit_param, reference_id: @reference.id)
         return render :unsubmitted unless @submit_reference_form.valid?
 
-        # TODO: Refactor the below into one form object so we don't need 5 condtionals
-        # Will be done in a follow up PR
-        if @submit_reference_form.submit == 'yes'
-          if Reference::RefereeTypeForm.build_from_reference(@reference).valid? &&
-              Reference::RefereeNameForm.build_from_reference(@reference).valid? &&
-              Reference::RefereeRelationshipForm.build_from_reference(@reference).valid? &&
-              Reference::RefereeEmailForm.build_from_reference(@reference).valid?
-
-            # TODO: Link up with Steve's work
-          else
-            # TODO: Once the forms are moved into objects this won't be necessary
-            # It can just call valid? and render the errors.
-            flash[:warning] = 'You must complete all of your referees details before your reference request can be submitted'
-
-            render :unsubmitted
-          end
+        if @submit_reference_form.send_request?
+          redirect_to candidate_interface_decoupled_references_create_candidate_name_path(@reference.id)
```

Comment:
Does this work as the `create` action is a post?

Reply:
Good point. As discussed i've added some logic to the new action. The duplication is taken care of by Malcolm's PR 
![image](https://user-images.githubusercontent.com/42515961/95741095-941c5300-0c85-11eb-92c1-b675955f90ed.png)

Reply:
I reckon we should do those checks on CandidateNameForm as part of ReviewController#submit (can be refactored into a private method). 

Something like:
```
if @reference_candidate_name_form.valid?
  CandidateInterface::DecoupledReferences::RequestReference.call(@reference, flash)
else
  redirect_to candidate_interface_decoupled_references_new_candidate_name_path
end
```

That way we aren't triggering side effects via a GET request.

Reply:
Have done 👍